File: 20151007/src/beginner_tutorials/scripts/mykeyop.py
#!/usr/bin/env python
import rospy
from kobuki_msgs.msg import MotorPower
from geometry_msgs.msg import Twist

def talker():
    pow_pub = rospy.Publisher('mobile_base/commands/motor_power', MotorPower)
    vel_pub = rospy.Publisher('mobile_base/commands/velocity', Twist);
    rospy.init_node('mykeyop')
    while not rospy.is_shutdown():
        motor_power = MotorPower()
        motor_power.state = motor_power.ON
        velocity = Twist()
        velocity.linear.x = 0.5
        velocity.angular.z = 0.3

        rospy.loginfo(motor_power)
        rospy.loginfo(velocity)
        pow_pub.publish(motor_power)
        vel_pub.publish(velocity)


        rospy.sleep(0.01)

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

File: 20151007/src/beginner_tutorials/src/mykey_op.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"
#include <iostream>

int main(int argc, char** argv)
{
	ros::init(argc, argv, "mykey_op");
	ros::NodeHandle n;
	ros::Publisher com_velocity = n.advertise<geometry_msgs::Twist>("mobile_base/commands/velocity", 100);
	ros::Rate loop_rate(1);
	geometry_msgs::Twist twist;
	std::cout << "Input linear param" << std::endl;
	std::cin >> twist.linear.x;
	std::cin >> twist.linear.y;
	std::cin >> twist.linear.z;
	std::cout << "Input angular param" << std::endl;
	std::cin >> twist.angular.x;
	std::cin >> twist.angular.y;
	std::cin >> twist.angular.z;
	while (ros::ok()) {
		com_velocity.publish(twist);
		ros::spinOnce();
		loop_rate.sleep();
	}

	return 0;
}
